src/plugins/data/common/expressions/interpreter_provider.ts

```diff
@@ -17,19 +17,25 @@
  * under the License.
  */
 
-import clone from 'lodash.clone';
-import { each, keys, last, mapValues, reduce, zipObject } from 'lodash';
-import { getType, fromExpression, getByAlias, castProvider } from '@kbn/interpreter/common';
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { clone, each, keys, last, mapValues, reduce, zipObject } from 'lodash';
+
+// @ts-ignore
+import { fromExpression, getType, getByAlias, castProvider } from '@kbn/interpreter/target/common';
+
 import { createError } from './create_error';
+import { ExpressionAST } from './types';
+
+export { createError };
 
-export function interpreterProvider(config) {
```

Comment:
This file took a ton of `any`s to get working in TS, do you think it’s worth it? What do you think @clintandrewhall? Do you have a plan for TS-isfying this code?

Reply:
> do you think it’s worth it?

If it's worth converting it to TypeScript? Definitely. Event just setting `any` everywhere, TypeScript is still more useful.

> Do you have a plan for TS-isfying this code?

Yes, the main thing that needs types in this file is `fn` and the should be easy (maybe we even have it already). And then `config` and `context` should be doable, too.

Though the fact that this function has word "provider" in it and that it uses so many things from `lodash` and does not have tests suggests that this file is a good candidate for some refactoring.

__

BTW, @spalger I've heard about a JS -> TS migration strategy where some automated codemod

1. renames all files `.js` to `.ts`
2. fixes basic errors
3. puts in `any`s everywhere it cannot fix
4. human fixes the rest

That way whole codebase is converted to `.ts` pretty much immediately. It has plenty `any`s though the code that was automatically converted, but the idea is: it is better to have TypeScript code with `any`s then JS code. And also that the "migration" that developers have to do is to improve the types, instead of chore like renaming the file names and fixing imports.

Reply:
> the idea is: it is better to have TypeScript code with `any`s then JS code.

I see the logic there, but I don't think we have an issue with mixing JS and TS code, except that it creates pressure to migrate JS to TS as we work in different parts of the code base, which is why I would prefer that we kept with the current strategy. If we did an auto-migration we would remove a lot of the pressure to properly ts-ify things properly, which doesn't seem like a good thing to me.

Also, it would really lower the bar for what our TS should look like, which I think would be harmful for the number of Kibana contributors who are still working in JS and putting off learning TS. Those people probably aren't too excited about learning TS and if the standard is "just put an `any` on it" then I think it'd create more problems than is solves.

Reply:
another problem with `any` - I often don't understand it is just a temporary mock or I can use `any` for real